mysql2 gem 0.3.18 fails to compile native extension on Rubinius 2.5.6

mysql2 gem 0.3.18 fails to compile the native extensions on Rubinius 2.5.6 on both mac (yosemite) and ubuntu 12.04.5 LTS.
It compiles correctly on rbx 2.5.5

Here's the error
```
$ ruby -v
rubinius 2.5.6 (2.1.0 57cc497d 2015-06-13 3.5.1 JI) [x86_64-darwin14.3.0]
Filipes-MBP-2:jmeter fmfdias$ gem uninstall pg
Successfully uninstalled pg-0.18.2
Filipes-MBP-2:jmeter fmfdias$ gem install mysql2
Fetching: mysql2-0.3.18.gem (100%)
Building native extensions.  This could take a while...
ERROR:  Error installing mysql2:
	ERROR: Failed to build gem native extension.

    /Users/fmfdias/.rubies/rbx-2.5.6/bin/rbx -r ./siteconf20150615-69648-1p123j5.rb extconf.rb
checking for ruby/thread.h... yes
checking for rb_thread_call_without_gvl() in ruby/thread.h... no
checking for rb_thread_blocking_region()... no
checking for rb_wait_for_single_fd()... no
checking for rb_hash_dup()... no
checking for rb_intern3()... no
-----
Using mysql_config at /usr/local/bin/mysql_config
-----
checking for mysql.h... yes
checking for errmsg.h... yes
checking for mysqld_error.h... yes
-----
Setting rpath to /usr/local/Cellar/mysql-connector-c/6.1.6/lib
-----
creating Makefile

make "DESTDIR=" clean

make "DESTDIR="
compiling client.c
client.c:23:14: error: static declaration of 'rb_hash_dup' follows non-static declaration
static VALUE rb_hash_dup(VALUE other) {
             ^
/Users/fmfdias/.rubies/rbx-2.5.6/vm/include/capi/ruby/ruby.h:1438:11: note: previous declaration is here
  VALUE   rb_hash_dup(VALUE self);
          ^
client.c:502:15: warning: expression result unused [-Wunused-value]
  RB_GC_GUARD(current);
              ^~~~~~~
client.c:686:17: warning: expression result unused [-Wunused-value]
    RB_GC_GUARD(inspect);
                ^~~~~~~
client.c:654:15: warning: expression result unused [-Wunused-value]
  RB_GC_GUARD(current);
              ^~~~~~~
client.c:1081:15: warning: expression result unused [-Wunused-value]
  RB_GC_GUARD(current);
              ^~~~~~~
4 warnings and 1 error generated.
make: *** [client.o] Error 1

make failed, exit code 2

Gem files will remain installed in /Users/fmfdias/.gem/rbx/2.5.6/gems/mysql2-0.3.18 for inspection.
Results logged to /Users/fmfdias/.gem/rbx/2.5.6/extensions/x86_64-darwin-14/2.1/mysql2-0.3.18/gem_make.out
```

The issue seems to have surfaced now because the rb_hash_dup function was added to this Rubinius version without defining HAVE_RB_HASH_DUP.
Nevertheless, the issue doesn't occur with mysql2 gem master branch since it doesn't have the offending static keyword anymore.

I'm going to open up a pull request where the HAVE_RB_HASH_DUP is defined and the mysql2 gem compiles. But since I don't know if this is the correct solution and the master mysql2 master branch compiles, I'm also opening the issue there so the best solution can be picked.

Thank you

--- a/vm/include/capi/ruby/ruby.h
+++ b/vm/include/capi/ruby/ruby.h
@@ -1436,9 +1436,11 @@ struct RTypedData {
 
   /** Duplicate the Hash object */
   VALUE   rb_hash_dup(VALUE self);
+#define HAVE_RB_HASH_DUP 1
 
   /** Freeze the Hash object */
   VALUE   rb_hash_freeze(VALUE self);
+#define HAVE_RB_HASH_FREEZE 1
 
   /** Return the value associated with the key, including default values. */
   VALUE   rb_hash_aref(VALUE self, VALUE key);

